### DISCUSSION TITLE
airobo_gazebo gazebo패키지에서_world.launch.py 실행시 Error [Param.cc:449] 발생

### DISCUSSION THREAD

**Discussion by eking158** [Road-Balance/gcamp_ros2_basic#25] [Q&A] [answered]
사용 os: ubuntu 20.04
사용 ros 버전: ros2 foxy

airobo_gazebo gazebo패키지에서 launch 명령어로_world.launch.py 실행시 '[gzserver-1] Error [Param.cc:449] Invalid argument. Unable to set value [world ] for key[odometry_source].' 해당 오류가 발생하고, gazebo 환경이 사진처럼 일부 누락된 상태로 불러와지는 문제가 발생합니다.
![스크린샷, 2022-07-12 13-42-13](https://user-images.githubusercontent.com/96982279/178410360-e61f3e68-d93f-454c-8b64-4818912070bc.png)

**Comment by eking158** [comment, marked as answer]
ros2 launch airobo_gazebo gazebo_world.launch.py 실행시 나오는 출력문 전문 입니다.


[INFO] [launch]: All log files can be found below /home/eking/.ros/log/2022-07-12-13-39-38-908945-eking-Ubuntu-25409
[INFO] [launch]: Default logging verbosity is set to INFO
 If it's your 1st time to download Gazebo model on your computer, it may take few minutes to finish. 
[INFO] [gzserver-1]: process started with pid [25411]
[INFO] [gzclient   -2]: process started with pid [25413]
[INFO] [robot_state_publisher-3]: process started with pid [25416]
[INFO] [joint_state_publisher-4]: process started with pid [25419]
[INFO] [spawn_entity.py-5]: process started with pid [25421]
[robot_state_publisher-3] Parsing robot urdf xml string.
[robot_state_publisher-3] Link chassis had 9 children
[robot_state_publisher-3] Link back_left_wheel had 0 children
[robot_state_publisher-3] Link back_right_wheel had 0 children
[robot_state_publisher-3] Link camera had 0 children
[robot_state_publisher-3] Link front_left_wheel had 0 children
[robot_state_publisher-3] Link front_right_wheel had 0 children
[robot_state_publisher-3] Link hokuyo had 0 children
[robot_state_publisher-3] Link imu_link had 0 children
[robot_state_publisher-3] Link left_track had 0 children
[robot_state_publisher-3] Link right_track had 0 children
[robot_state_publisher-3] [INFO] [1657600779.439151922] [robot_state_publisher]: got segment back_left_wheel
[robot_state_publisher-3] [INFO] [1657600779.439283668] [robot_state_publisher]: got segment back_right_wheel
[robot_state_publisher-3] [INFO] [1657600779.439290211] [robot_state_publisher]: got segment camera
[robot_state_publisher-3] [INFO] [1657600779.439294330] [robot_state_publisher]: got segment chassis
[robot_state_publisher-3] [INFO] [1657600779.439299641] [robot_state_publisher]: got segment front_left_wheel
[robot_state_publisher-3] [INFO] [1657600779.439304451] [robot_state_publisher]: got segment front_right_wheel
[robot_state_publisher-3] [INFO] [1657600779.439309311] [robot_state_publisher]: got segment hokuyo
[robot_state_publisher-3] [INFO] [1657600779.439313961] [robot_state_publisher]: got segment imu_link
[robot_state_publisher-3] [INFO] [1657600779.439319062] [robot_state_publisher]: got segment left_track
[robot_state_publisher-3] [INFO] [1657600779.439323802] [robot_state_publisher]: got segment right_track
[robot_state_publisher-3] [INFO] [1657600779.439328692] [robot_state_publisher]: got segment robot_footprint
[spawn_entity.py-5] [INFO] [1657600779.761219573] [spawn_entity]: Spawn Entity started
[spawn_entity.py-5] [INFO] [1657600779.761663152] [spawn_entity]: Loading entity published on topic robot_description
[spawn_entity.py-5] [INFO] [1657600779.763066551] [spawn_entity]: Waiting for entity xml on robot_description
[joint_state_publisher-4] [INFO] [1657600779.811090691] [joint_state_publisher]: Waiting for robot_description to be published on the robot_description topic...
[spawn_entity.py-5] [INFO] [1657600779.869664517] [spawn_entity]: Waiting for service /spawn_entity, timeout = 30
[spawn_entity.py-5] [INFO] [1657600779.870650613] [spawn_entity]: Waiting for service /spawn_entity
[spawn_entity.py-5] [INFO] [1657600780.627357460] [spawn_entity]: Calling service /spawn_entity
[gzserver-1] [WARN] [1657600893.604140900] [rcl]: Found remap rule 'camera/image_raw:=/skidbot/camera/image'. This syntax is deprecated. Use '--ros-args --remap camera/image_raw:=/skidbot/camera/image' instead.
[gzserver-1] [WARN] [1657600893.604262466] [rcl]: Found remap rule 'camera/camera_info:=/skidbot/camera/image/camera_info'. This syntax is deprecated. Use '--ros-args --remap camera/camera_info:=/skidbot/camera/image/camera_info' instead.
[gzserver-1] [WARN] [1657600893.608678617] [rcl]: Found remap rule 'camera/image_raw:=/skidbot/camera/image'. This syntax is deprecated. Use '--ros-args --remap camera/image_raw:=/skidbot/camera/image' instead.
[gzserver-1] [WARN] [1657600893.608734557] [rcl]: Found remap rule 'camera/camera_info:=/skidbot/camera/image/camera_info'. This syntax is deprecated. Use '--ros-args --remap camera/camera_info:=/skidbot/camera/image/camera_info' instead.
[gzserver-1] [INFO] [1657600893.617027643] [skidbot.camera_controller]: Publishing camera info to [/skidbot/camera_sensor/camera_info]
[spawn_entity.py-5] [INFO] [1657600893.637111392] [spawn_entity]: Spawn status: SpawnEntity: Successfully spawned entity [fusionbot]
[INFO] [spawn_entity.py-5]: process has finished cleanly [pid 25421]
[gzserver-1] [WARN] [1657600894.031633489] [rcl]: Found remap rule '~/out:=scan'. This syntax is deprecated. Use '--ros-args --remap ~/out:=scan' instead.
[gzserver-1] [WARN] [1657600894.035278811] [rcl]: Found remap rule '~/out:=scan'. This syntax is deprecated. Use '--ros-args --remap ~/out:=scan' instead.
[gzserver-1] [INFO] [1657600894.046592437] [skidbot.imu.imu_plugin]: <initial_orientation_as_reference> is unset, using default value of false to comply with REP 145 (world as orientation reference)
[gzserver-1] [WARN] [1657600894.160159286] [rcl]: Found remap rule '/cmd_vel:=cmd_vel'. This syntax is deprecated. Use '--ros-args --remap /cmd_vel:=cmd_vel' instead.
[gzserver-1] [WARN] [1657600894.160191353] [rcl]: Found remap rule '/odom:=odom'. This syntax is deprecated. Use '--ros-args --remap /odom:=odom' instead.
[gzserver-1] [INFO] [1657600894.164465868] [skidbot.skid_steer_drive_controller]: Wheel pair 1 separation set to [0.400000m]
[gzserver-1] [INFO] [1657600894.164505084] [skidbot.skid_steer_drive_controller]: Wheel pair 2 separation set to [0.400000m]
[gzserver-1] [INFO] [1657600894.164513246] [skidbot.skid_steer_drive_controller]: Wheel pair 1 diameter set to [0.200000m]
[gzserver-1] [INFO] [1657600894.164520206] [skidbot.skid_steer_drive_controller]: Wheel pair 2 diameter set to [0.200000m]
[gzserver-1] [WARN] [1657600894.164572212] [rcl]: Found remap rule '/cmd_vel:=cmd_vel'. This syntax is deprecated. Use '--ros-args --remap /cmd_vel:=cmd_vel' instead.
[gzserver-1] [WARN] [1657600894.164579372] [rcl]: Found remap rule '/odom:=odom'. This syntax is deprecated. Use '--ros-args --remap /odom:=odom' instead.
[gzserver-1] [INFO] [1657600894.165841331] [skidbot.skid_steer_drive_controller]: Subscribed to [/skidbot/cmd_vel]
[gzserver-1] Error [Param.cc:449] Invalid argument. Unable to set value [world ] for key[odometry_source].
[gzserver-1] [INFO] [1657600894.167050123] [skidbot.skid_steer_drive_controller]: Advertise odometry on [/skidbot/odom]
[gzserver-1] [INFO] [1657600894.168080606] [skidbot.skid_steer_drive_controller]: Publishing odom transforms between [odom] and [robot_footprint]
[gzserver-1] [INFO] [1657600894.168114515] [skidbot.skid_steer_drive_controller]: Publishing wheel transforms between [robot_footprint], [front_left_wheel_joint] and [front_right_wheel_joint]
[gzserver-1] [INFO] [1657600894.168121646] [skidbot.skid_steer_drive_controller]: Publishing wheel transforms between [robot_footprint], [back_left_wheel_joint] and [back_right_wheel_joint]

**Reply by kimsooyoung** [reply]
3DGEMS 데이터셋을 잘 설정하셨는지 보세요

**Reply by eking158** [reply]
넵 해결 되었습니다. 감사합니다!
